crates/binstalk/src/ops/resolve/resolution.rs

```diff
@@ -87,14 +88,27 @@ impl ResolutionFetch {
             current_version: self.new_version,
             source: self.source,
             target: self.fetcher.target().to_compact_string(),
-            bins: self
-                .bin_files
-                .into_iter()
-                .map(|bin| bin.base_name)
-                .collect(),
+            bins: Self::resolve_bins(&opts.bins, self.bin_files),
         })
     }
 
+    fn resolve_bins(
+        user_specified_bins: &Option<Vec<CompactString>>,
+        crate_bin_files: Vec<BinFile>,
+    ) -> Vec<CompactString> {
+        // We need to filter crate_bin_files by user_specified_bins in case the prebuilt doesn't
+        // have featured-gated (optional) binary (gated behind feature).
+        crate_bin_files
+            .iter()
+            .map(|bin| bin.base_name.clone())
```

Comment:
```suggestion
        crate_bin_files
            .into_iter()
            .map(|bin| bin.base_name)
```

This clone here is unnecessary